lib/fuse_lowlevel.c

```diff
@@ -2019,7 +2019,7 @@ void do_init(fuse_req_t req, fuse_ino_t nodeid, const void *inarg)
 				bufsize = max_bufsize;
 			}
 		}
-		if (arg->minor >= 38)
+		if (arg->minor >= 38 || (inargflags & FUSE_HAS_EXPIRE_ONLY))
```

Comment:
Sorry, seems I somehow had forgotten to press the right button - this review comment was in "pending" state.

Reply:
it was first introduced by kernel 6.2? and there is no extra capability flag to ask the kernel if it supports it. you only know it from the header version.

if i remember correctly in the kernel, this change starts using a previously declared "padding" area. if you do not check if the kernel supports it you do an `invalidate entry` instead of an `expire entry`.

Reply:
@HereThereBeDragons look further above this function

```
		inargflags = arg->flags;
		if (inargflags & FUSE_INIT_EXT)
			inargflags = inargflags | (uint64_t) arg->flags2 << 32;
```

FUSE_HAS_EXPIRE_ONLY is (1ULL << 35>), so is in the high bits, which is only set when FUSE_INIT_EXT. Otherwise we couldn't trust these flags at all. Using the version is really not appropriate, because it would make back porting impossible. In fact, I have back ported some features to older kernels with a fuse-dkms package - relying on the version number is just not possible.